File: 2021/day7/go/main_test.go
package main

import (
	"testing"
)

var input = []string{
	"16",
	"1",
	"2",
	"0",
	"4",
	"2",
	"7",
	"1",
	"2",
	"14",
}

func TestPart1(t *testing.T) {
	expected := 37
	actual := part1(input)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

func TestPart2(t *testing.T) {
	expected := 168
	actual := part2(input)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

File: 2022/day10/js/main.test.js
"use strict";

const assert = require("assert");
const fs = require("fs");
const { part1, part2 } = require("./main");

async function test_part1(input) {
  const expected = 13140;
  const actual = await part1(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test_part2(input) {
  const expected = 0;
  const actual = await part2(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test() {
  const data = fs.readFileSync("../input.test.txt", { encoding: "utf8" });
  const input = data.split("\n");
  input.reverse();
  await test_part1(input);
  await test_part2(input);
}

test();
